service/storage_service.cc

```diff
@@ -365,6 +370,21 @@ void storage_service::prepare_to_join(std::vector<inet_address> loaded_endpoints
         }
     }
 
+    if (replacing_a_node_with_same_ip) {
+        slogger.info("Replacing a node with same IP address, my address={}, node being replaced={}",
+                    get_broadcast_address(), get_broadcast_address());
+        slogger.info("Update tokens for replacing node early, replacing node has the same IP address of the node being replaced");
+        _token_metadata.update_normal_tokens(_bootstrap_tokens, get_broadcast_address());
```

Comment:
This place could use a comment explaining why we need to update tokens early in this case.

The commit message mentions a hash. How is the hash calculated? What is it used for?

Reply:
It is prepare patch for "gossip: Make boot process faster and safer". We want the replacing node and the rest of the cluster has the same view the normal tokens, the hash is the hash of the normal tokens.

Reply:
> We want the replacing node and the rest of the cluster has the same view the normal tokens

Great - deserves a comment above the `if`.

Reply:
done